### PR TITLE
name filter added

### PR DESCRIPTION
**Pull request by ChitvanRamani22**
Name filter added.
@Namanl2001 @sagar-barapatre please review this.

### CODE REVIEW

**Comment by Namanl2001** [comment]
Would you mind resolving merge conflicts, Please?

**Comment by ChitvanRamani22** [comment]
@Namanl2001 Done!

File: src/components/filter.jsx [hunk 1]
```diff
@@ -1,59 +1,71 @@
...
+              className='textbox'
+              type='number'
+              onChange={this.handlePinChange}
+              value={this.state.pin}
+              placeholder='Search by Pin-Code'
+              mr='sm-9'
+              min='111111'
+              max='999999'
+            />
+            <input title='Search' value='' type='submit' className='button' />
```
Comment (Namanl2001) [inline src/components/filter.jsx:54]: What is assigned to `value` ?

File: src/components/filter.jsx [hunk 2]
```diff
@@ -1,59 +1,71 @@
...
+          <form onSubmit={this.handleSubmitName} className='form2'>
+            <input
+              className='textbox'
+              type='string'
+              onChange={this.handleNameChange}
+              value={this.state.name}
+              placeholder='Search by Name'
+              mr='sm-9'
+            />
+            <input title='Search' value='' type='submit' className='button' />
```
Comment (Namanl2001) [inline src/components/filter.jsx:65]: Here also

File: src/store/reducer/user.js [hunk 3]
```diff
@@ -141,6 +145,18 @@ function rootReducer(state = defaultState, action) {
         users: newArray4,
         class: action.type.slice(1),
       };
+    case 'NAME':
+      const newArray8 = state.pindata.filter(
```
Comment (Namanl2001) [inline src/store/reducer/user.js:149]: Now acc. to sequence this should be `newArray5` ig.

**Review by Namanl2001 — CHANGES_REQUESTED** [review]
Small nits else, LGTM!

File: src/store/reducer/user.js (hunk 3 above)
Comment (ChitvanRamani22) [reply src/store/reducer/user.js]: I have changed it to 5.

File: src/components/filter.jsx (hunk 1 above)
Comment (ChitvanRamani22) [reply src/components/filter.jsx:54]: This value is used to display the search symbol.

**Comment by ChitvanRamani22** [comment]
@Namanl2001 i have made the changes.

**Review by Namanl2001 — APPROVED** [review]
Nice work! Thanks @ChitvanRamani22

**Pull request merged**